File: src/main/java/ast/ASTVarExpr.java
package ast;

public class ASTVarExpr extends ASTExpr {
    String name;

    public ASTVarExpr(ASTFileLocation loc, String name){
        super(loc);
        this.name = name;
    }

    public void print(int i){
        printIndent(i);
        System.out.print("ASTVarExpr: ");
        System.out.println(name);

    }
}

File: src/main/kotlin/ast/ASTVarExpr.kt
package ast

class ASTVarExpr(loc: ASTFileLocation, val name: String) : ASTExpr(loc) {

    override fun isLValue(): Boolean {
        return true
    }

    override fun print(i: Int) {
        printIndent(i)
        print("ASTVarExpr: ")
        println(name)

    }
}